the-collab-lab/tcl-69-smart-shopping-list
# [JH] Restrict Route Access

## Description

This code restricts route access to "List" and "Manage List" when user is not signed in. Using the `<Navigate>` component, user is redirected to "Home" or "/" route path to sign in. 

Note: The route path to "Home" currently renders information about user's lists. This may be changed to the landing page in a future issue.  

## Related Issue

Closes #51 
Sub-issue of #14 

## Acceptance Criteria

- [x] Users who are not signed in won't be able to navigate to "List" and "Manage List" pages via navbar or directly through the URL.

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

## Updates

### Before
Successfully navigated to "Manage List" page without signing in.
<img width="1262" alt="Screen Shot 2024-03-21 at 2 17 33 PM" src="https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/64502078/6c915700-935e-42f3-bdcb-e38fdca59d72">

### After
Page redirected to "Home" when clicking on "Manage List."
<img width="1267" alt="Screen Shot 2024-03-21 at 2 20 35 PM" src="https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/64502078/3ae6716a-88b3-4bd5-8fe8-cf50d917ec16">

## Testing Steps / QA Criteria
1. Without signing in, try navigate to "List" page using the navbar or direct URL (/list). Page should redirect to "Home."
2. Repeat for "Manage List."
